### PR TITLE
bump aws dep version

### PR DESCRIPTION
**Pull request by skyzh**
## Describe your changes

This PR is simply the patch from https://github.com/neondatabase/neon/issues/4008 except we enabled `force_path_style` for custom endpoints. This is because at some version, the s3 sdk by default uses the virtual-host style access, which is not supported by MinIO in the default configuration. By enforcing path style access for custom endpoints, we can pass all e2e test cases.

SDK 0.55 is not the latest version and we can bump it further later when all flaky tests in this PR are resolved.

This PR also (hopefully) fixes flaky test `test_ondemand_download_timetravel`.

## Issue ticket number and link

close https://github.com/neondatabase/neon/issues/4008

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- ~[ ] If it is a core feature, I have added thorough tests.~
- ~[ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?~
- ~[ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.~

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by skyzh** [comment]
using the deprecated `Endpoint` API does not work and will cause all create_timeline request to timeout, and I guess there are some internal changes in AWS SDK that caused the failure...

**Comment by skyzh** [comment]
e2e tests are working (in neon-cloud) but there are some regression tests failing, which I believe is not really related to this upgrade...?

**Comment by skyzh** [comment]
I believe the root cause of the previous aws-sdk-s3 upgrade failure is that, the SDK now supports the virtual-host style access pattern (i.e., bucket.region.aws.com), and the MinIO instances do not support that by default unless we configure it. Therefore, by forcing path-style accessing only for custom endpoints, we can make it compatible with MinIO.

File: test_runner/regress/test_ondemand_download.py [hunk 1]
```diff
@@ -225,7 +225,7 @@ def get_resident_physical_size():
     assert filled_current_physical == filled_size, "we don't yet do layer eviction"
 
     # Wait until generated image layers are uploaded to S3
-    time.sleep(3)
+    time.sleep(5)
```
Comment (hlinnaka) [inline test_runner/regress/test_ondemand_download.py:228]: This seems unrelated to aws sdk version, was this test flaky?

Perhaps we could use `wait_for_upload_queue_empty` here. Looking at the git history, `wait_for_upload_queue_empty` didn't exist when this sleep was introduced.

**Review by hlinnaka — APPROVED** [review]

**Pull request merged**